project_status.py

```diff
@@ -215,108 +211,449 @@ def parse_qualimap(genome_results_file, sample):
+    return result
+
+def get_samples_under_qc(project):
+    result = []
+    command = "jobinfo | grep qc_{}".format(project)
+    try:
+        p = subprocess.Popen(command, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+        output = p.communicate()
+    except Exception, e:
+        print "Cannot execute command {}".format(command)
+        raise e
+    output = output[0].split('\n')
+    for line in output:
+        # skip empty lines
+        if line.strip() != '':
+            sample = line.split('qc_{}-'.format(project))[-1]
+            if sample not in result:
+                result.append(sample)
+    return result
+
+def get_samples_with_failed_analysis(project, analysis_dir):
+    log_path = os.path.join(analysis_dir, '{}/piper_ngi/logs/{}-*.exit'.format(project, project))
+    under_analysis = get_samples_under_analysis(project)
+    exit_files = glob.glob(log_path)
+    result = {}
+    for path in exit_files:
+        with open(path, 'r') as exit_file:
+            exit_code = exit_file.read().strip()
+            # P4603-P4603_189-merge_process_variantcall.exit
+            sample = os.path.basename(path).replace('{}-'.format(project), '').split('-')[0]
+            if exit_code == '' and sample not in under_analysis:
+                if sample in result:
+                    result[sample] = [result[sample]]
+                    result[sample].append('Empty exit code, but sample is not under analysis')
+                else:
+                    result[sample] = 'Empty exit code, but sample is not under analysis'
+            elif exit_code.strip() != '0' and exit_code != '':
+                if sample in result:
+                    result[sample] = [result[sample]]
+                    result[sample].append('Exit code: {}'.format(exit_code))
+                else:
+                    result[sample] = 'Exit code: {}'.format(exit_code)
+    return result
+
+def get_incoherent_samples(results_francesco):
+    result = {}
+    for sample_id, sample in results_francesco.items():
+        sequenced = sample.get('#Archived_runs', '')
+        organized = sample.get('#Data_runs', '')
+        analyzed = sample.get('#Analysis_runs', '')
+        try:
+            sequenced = int(sequenced)
+            organized = int(organized)
+            analyzed = int(analyzed)
+        # if not int or something strange in the results, print it too (just in case)
+        except ValueError, e:
+            result[sample_id] = {'sequenced': sequenced, 'organized': organized, 'analyzed': analyzed}
+        else:
+            if not(sequenced == organized == analyzed):
+                result[sample_id] = {'sequenced': sequenced, 'organized': organized, 'analyzed': analyzed}
+    return result
+
+
+def get_sequenced(project):
+    incoming = "/proj/ngi2016003/incoming"
+    project_flowcells = {}
+    for fc in os.listdir(incoming):
+        sample_sheet = os.path.join(incoming, fc, 'SampleSheet.csv')
```

Comment:
for example `find_samples_from_archive ` does not assume the presence of SampleSheet.csv . There is no need to assume that this file exists